ion-schema/src/system.rs

```diff
@@ -862,27 +929,65 @@ impl Resolver {
         // get all isl type names that are defined within the schema
         // this will be used to resolve type references which might not have yet resolved while loading a type definition
         let isl_type_names: HashSet<&str> = HashSet::from_iter(
-            isl.types()
+            isl_types
                 .iter()
-                .filter(|t| matches!(t, IslType::Named(_)))
+                .filter(|t| matches!(t, IslTypeKind::Named(_)))
                 .map(|t| t.name().as_ref().unwrap().as_str()),
         );
 
         // Resolve all ISL types and constraints
         for isl_type in isl.types() {
             let pending_types = &mut PendingTypes::default();
 
-            match isl_type {
-                IslType::Named(named_isl_type) => {
+            match &isl_type.kind {
+                IslTypeKind::Named(named_isl_type) => {
+                    match isl_version {
```

Comment:
The match arms here are nearly identical. It should be possible to get rid of the match entirely using something like this: 
```
isl_type.constraints().iter().any(|c| c.version != isl_version)
```

Reply:
The reason I was doing this is because I am using v_1_0 constraints methods for cosntraints that are working identically in both versions. But I see how this can get complex to check when we have more ISL versions in future. 

I will change the places where ISL constraints for ISL 2.0 are using ISL 1.0 construction methods and rather use its own which passes correct version.

```rust
   /// current approach
    pub fn all_of<A: Into<Vec<IslTypeRef>>>(isl_types: A) -> IslConstraint {
        v_1_0::all_of(isl_types)
    }
    
    // new approach
     pub fn all_of<A: Into<Vec<IslTypeRef>>>(isl_types: A) -> IslConstraint {
     // the following is similar to what v_!_0 is doign except the version detail
        IslConstraint::new(
            IonSchemaLanguageVersion::V2_0,
            IslConstraintImpl::AllOf(
                isl_types
                    .into()
                    .into_iter()
                    .map(|t| t.type_reference)
                    .collect(),
            ),
        )
    }


```